File: ordermanager.client/src/models/Employee.ts


// src/types/Employee.ts

export interface Employee {
    employeeId: number;
    firstName: string;
    middleInitial: string;
    lastName: string;
  }
  
  export interface EmployeeDetails extends Employee {
    orderCount: number;
  
    // This is not returned by the API — add it client-side
    displayName?: string; // optional — derived field
  }
  

File: OrderManager.Server/Models/Employee.cs

namespace OrderManager.Server.Models
{
    //Maps to the Employee table in the database
    public class Employee
    {
        public int EmployeeId { get; set; }
        public string FirstName { get; set; } = string.Empty;
        public string MiddleInitial { get; set; } = string.Empty;
        public string LastName { get; set; } = string.Empty;

    }

    // Used to return additional details about the employee on the API
    public class EmployeeDetails : Employee
    {
        public int OrderCount { get; set; }
        public string DisplayName => $"{FirstName} {MiddleInitial} {LastName}";
    }
}
